### PR TITLE
support decimal type

### PR DESCRIPTION
**Pull request by EagleChen**

### CODE REVIEW

**Comment by EagleChen** [comment]
@siddontang 
Can you review & merge this commit and update the reference in "go-mysql-elasticsearch"?

**Comment by siddontang** [comment]
Hi @EagleChen , now we only use float64 for mysql decimal type, it is not ok and we may lose precision. 
Maybe we can use a decimal package to handle it?

**Comment by EagleChen** [comment]
@siddontang 
Sorry, I don't fully understand your explanation. Maybe you misunderstand the purpose of this pull request.

From my aspect, `decimal` is a type that mysql supports. It's probably users may use it to define some fields ( which you can see in the tests I've updated in this pull request ).

For now, if a mysql table contains a field of type `decimal`, then this tool fails to dump the data ( [errors](https://github.com/siddontang/go-mysql/blob/3f9fa3bf1fd4d673fc97494a21c8251725bd50c4/canal/dump.go#L56-L57) ).

**Comment by siddontang** [comment]
@EagleChen  what I mean is that now the decimal support may lose precision because I use float64 in go, so it may be a risk. :-) 

maybe later I can find a go decimal package and use it for mysql decimal type.

File: schema/schema.go [hunk 1]
```diff
@@ -52,7 +52,7 @@ func (ta *Table) AddColumn(name string, columnType string, extra string) {
 
 	if strings.Contains(columnType, "int") || strings.HasPrefix(columnType, "year") {
 		ta.Columns[index].Type = TYPE_NUMBER
-	} else if columnType == "float" || columnType == "double" {
+	} else if columnType == "float" || columnType == "double" || strings.HasPrefix(columnType, "decimal") {
```
Comment (siddontang) [inline schema/schema.go]: can you help me use strings.HasPrefix to check float and double too. the double can be defined like "double(4,2)" too.

**Comment by EagleChen** [comment]
@siddontang 
Oh, OK.
Hope you implement this ASAP. :-)

File: schema/schema.go (hunk 1 above)
Comment (EagleChen) [inline schema/schema.go]: @siddontang  No problem. I will update this pull request very soon

**Comment by EagleChen** [comment]
@siddontang  pull request updated.

**Comment by siddontang** [comment]
Then you can update the godep in go-mysql-elasticsearch :-)

**Comment by EagleChen** [comment]
@siddontang  Pull request has been sent to go-mysql-elasticsearch :-)

**Pull request merged**